lib/spack/spack/binary_distribution.py

```diff
@@ -2517,6 +2532,10 @@ def _ensure_common_prefix(tar: tarfile.TarFile) -> str:
 
 
 def install_root_node(spec, unsigned=False, force=False, sha256=None):
+    # TODO?: Go through this and determine where it needs to be the deploy_spec vs. the spec.
+    # TODO?: Modify dependency lookup s.t. if it's not there, figure out what virtual it provided
+    # and find out what is now providing that virtual. Else raise an error that it didn't exist.
+    # Inverts the logic where you would call the build_spec as spec and the spec as deploy_spec
```

Comment:
Does this still need to be addressed?

Reply:
I don't believe so. I think we discovered that the correct behavior fell out naturally once we got `relocate_package` modified